src/components/NewClusterModal.js

```diff
@@ -230,15 +239,54 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
             if (isSelectedImageInputted) {
               this.setState({ viewMode: 'warning' })
             } else if (!!currentCluster) {
-              this.setState({ viewMode: 'replace' })
+              this.setState({ viewMode: _.lowerCase(getUpdateOrReplace()) })
             } else {
               this.createCluster()
             }
           }
-        }, !!currentCluster ? 'Replace' : 'Create')
+        }, !!currentCluster ? getUpdateOrReplace() : 'Create')
       ])
     ])
 
+    //determines whether the changes are applicable for a call to the leo patch endpoint
+    //see this for a diagram of the conditional this implements https://drive.google.com/file/d/1mtFFecpQTkGYWSgPlaHksYaIudWHa0dY/view
+    //this function returns true for cases 2 & 3 in this diagram
+    const canUpdate = () => {
```

Comment:
this is a bit nasty, I'm sure it can be simplified a bit more, but I doubt it will ever be both concise and readable, so I chose readable.

Reply:
Whatever the final logic, expressing it like a simple functional computation like this is a good approach, and will enable us to read and/or change it later if needed.

Reply:
It's also worth mentioning that this can go away once [this](https://broadworkbench.atlassian.net/secure/RapidBoard.jspa?rapidView=35&projectKey=IA&view=planning&selectedIssue=IA-1507&issueLimit=100&search=allowDelete) ticket is implemented, but it hasn't been prioritized yet